test/src/main/java/org/apache/accumulo/test/ComprehensiveTableOperationsIT.java

```diff
@@ -0,0 +1,957 @@
+  }
+
+  @Test
+  public void
+      test_setProperty_modifyProperties_removeProperty_getProperties_getTableProperties_getConfiguration()
+          throws Exception {
+    // These may be tested elsewhere across several tests, but simpler to test all here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+    // system and user tables
+    Set<String> allTables = ops.list();
+    for (String tableName : allTables) {
+      // setProperty
+      String propKey = Property.TABLE_ARBITRARY_PROP_PREFIX.getKey() + "prop";
+      String propVal = "val";
+      String newPropVal = "newval";
+      ops.setProperty(tableName, propKey, propVal);
+      // getProperties, getTableProperties, and getConfiguration
+      assertTrue(propFound(tableName, propKey, propVal));
+      assertEquals(propVal, ops.getTableProperties(tableName).get(propKey));
+      assertEquals(propVal, ops.getConfiguration(tableName).get(propKey));
+      // modifyProperties
+      ops.modifyProperties(tableName, properties -> properties.put(propKey, newPropVal));
+      // getProperties, getTableProperties, and getConfiguration
+      assertTrue(propFound(tableName, propKey, newPropVal));
+      assertEquals(newPropVal, ops.getTableProperties(tableName).get(propKey));
+      assertEquals(newPropVal, ops.getConfiguration(tableName).get(propKey));
+      // removeProperty
+      ops.removeProperty(tableName, propKey);
+      // getProperties, getTableProperties, and getConfiguration
+      assertFalse(propFound(tableName, propKey, newPropVal));
+      assertNull(ops.getTableProperties(tableName).get(propKey));
+      assertNull(ops.getConfiguration(tableName).get(propKey));
+    }
+  }
+
+  @Test
+  public void test_setLocalityGroups_getLocalityGroups() throws Exception {
+    // setLocalityGroups, getLocalityGroups for user tables is tested in LocalityGroupIT. Ensure
+    // test exists
+    assertDoesNotThrow(() -> Class.forName(LocalityGroupIT.class.getName()));
+    // setLocalityGroups, getLocalityGroups for system tables not tested for. Test basic
+    // functionality here
+    for (var sysTable : AccumuloTable.values()) {
+      LocalityGroupIT.createAndSetLocalityGroups(client, sysTable.tableName());
+      LocalityGroupIT.verifyLocalityGroupSet(client, sysTable.tableName());
+    }
+  }
+
+  @Test
+  public void test_importDirectory() throws Exception {
+    // importDirectory for user tables is tested in BulkNewIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(BulkNewIT.class.getName()));
+    // importDirectory for system tables not tested for. Test basic functionality here
+    var rootPath = getCluster().getTemporaryPath().toString();
+    var dir = rootPath + "/" + getUniqueNames(1)[0];
+    BulkNewIT.writeData(dir + "/f1.", getCluster().getServerContext().getConfiguration(), 0, 5, 0);
+
+    for (var sysTable : AccumuloTable.values()) {
+      assertThrows(Exception.class, () -> ops.importDirectory(dir).to(sysTable.tableName()).load());
```

Comment:
Some throw AccumuloException and some throw AccumuloSecurityException not sure if that matters. The AccumuloException occurs from FateServiceHandler and the AccumuloSecurityException occurs from ClientServiceHandler. The validation done in FateServiceHandler only checks that it's not the ROOT table. Should probably check that it's not a built in table. This would result in the same Exception for all system tables and is more clear.

Reply:
For these different exceptions you are seeing, do they seem correct?  Like is the AccumuloException related to a not security issue and it the AccumuloSecurityException related to a security issue?

Reply:
With BULK_IMPORT2, the table is validated:
`final var tableId = validateTableIdArgument(arguments.get(0), tableOp, NOT_ROOT_TABLE_ID);`
Allowing tableIds that aren't the ROOT table to pass this check.
Then, the table is further validated to check if the user has permission to BULK_IMPORT2 into the table:
`canBulkImport = security.canBulkImport(c, tableId, tableName, dir, null, namespaceId);`

I think the validation of the tableId should include all system tables. I don't think we ever would want a user to bulk import to a system table, regardless of permissions.